File: app/src/main/java/movies/com/MovieAdapter.java
package movies.com;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.squareup.picasso.Picasso;

import java.util.List;

import movies.com.dbqr.R;
import movies.com.model.Movie;


public class MovieAdapter extends RecyclerView.Adapter<MovieAdapter.MovieViewHolder> {

    private List<Movie> MovieCellList;
    private ActionInterface mLisenner;


    public MovieAdapter(List<Movie> contactList, ActionInterface listenner) {
        this.MovieCellList = contactList;
        mLisenner = listenner;
    }

    @Override
    public int getItemCount() {
        return MovieCellList.size();
    }

    public Movie getItem(int position) {
        return MovieCellList.get(position);
    }


    @Override
    public MovieViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View horizontalItem = LayoutInflater.from(parent.getContext()).inflate(R.layout.contact_item, parent, false);
        return new MovieViewHolder(horizontalItem);
    }

    @Override
    public void onBindViewHolder(MovieViewHolder holder, int position) {

        final Movie movie = getItem(position);
        if (movie != null) {
            holder.nameView.setText(movie.getTitle());
            holder.yearView.setText(movie.getReleaseYear()+"");

            Picasso.with(holder.imgView.getContext()).load(movie.getImage())
                    .error(R.drawable.ic_launcher_background)
                    .into(holder.imgView);
        }
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mLisenner.goToDescription(movie);
            }
        });
    }

    public class MovieViewHolder extends RecyclerView.ViewHolder {


        public MovieViewHolder(View itemView) {
            super(itemView);
            nameView =  itemView.findViewById(R.id.name_text);
            yearView =  itemView.findViewById(R.id.year_text);
            imgView =  itemView.findViewById(R.id.img_view);

        }

        private final ImageView imgView;
        TextView nameView;
        TextView yearView;
    }
    public void updateList(List<Movie> newlist) {
        MovieCellList.clear();
        MovieCellList.addAll(newlist);
        this.notifyDataSetChanged();
    }
}

File: app/src/main/java/movies/com/MovieAdapter.kt
package movies.com

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.bumptech.glide.Glide
import movies.com.dbqr.R
import movies.com.model.Movie


class MovieAdapter(private val MovieCellList: MutableList<Movie>, private val mLisenner: ActionInterface) : RecyclerView.Adapter<MovieAdapter.MovieViewHolder>() {

    override fun getItemCount(): Int {
        return MovieCellList.size
    }

    fun getItem(position: Int): Movie? {
        return MovieCellList[position]
    }


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MovieViewHolder {
        val horizontalItem = LayoutInflater.from(parent.context).inflate(R.layout.contact_item, parent, false)
        return MovieViewHolder(horizontalItem)
    }

    override fun onBindViewHolder(holder: MovieViewHolder, position: Int) {

        val movie = getItem(position)
        if (movie != null) {
            holder.nameView.text = movie.title
            holder.yearView.text = movie.releaseYear.toString() + ""
            Glide.with(holder.imgView.getContext()).load(movie.image).into(holder.imgView);
        }
        holder.itemView.setOnClickListener { mLisenner.goToDescription(movie!!) }
    }

    inner class MovieViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        internal val imgView: ImageView = itemView.findViewById(R.id.img_view)
        internal var nameView: TextView = itemView.findViewById(R.id.name_text)
        internal var yearView: TextView = itemView.findViewById(R.id.year_text)

    }
}
